libs/server/PubSub/SubscribeBroker.cs

```diff
@@ -408,6 +410,157 @@ public unsafe void Publish(byte* key, byte* value, int valueLength, bool ascii =
+                    while (!RespWriteUtils.WriteEmptyArray(ref curr, end))
+                        ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+                    return;
+                }
+
+                if (input.parseState.Count == 0)
+                {
+                    while (!RespWriteUtils.WriteArrayLength(subscriptions.Count, ref curr, end))
+                        ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+
+                    foreach (var key in subscriptions.Keys)
+                    {
+                        while (!RespWriteUtils.WriteBulkString(key.AsSpan().Slice(sizeof(int)), ref curr, end))
+                            ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+                    }
+                    return;
+                }
+
+                var totalArrayHeaderLen = 0;
+                while (!RespWriteUtils.WriteArrayLength(subscriptions.Count, ref curr, end, out var _, out totalArrayHeaderLen))
+                    ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+
+                var noOfFoundChannels = 0;
+                var pattern = input.parseState.GetArgSliceByRef(0).SpanByte;
+                var patternPtr = pattern.ToPointer() - sizeof(int);
+                *(int*)patternPtr = pattern.Length;
+
+                foreach (var key in subscriptions.Keys)
+                {
+                    fixed (byte* keyPtr = key)
+                    {
+                        var endKeyPtr = keyPtr;
+                        var _patternPtr = patternPtr;
+                        if (keySerializer.Match(ref keySerializer.ReadKeyByRef(ref endKeyPtr), true, ref keySerializer.ReadKeyByRef(ref _patternPtr), true))
+                        {
+                            while (!RespWriteUtils.WriteSimpleString(key.AsSpan().Slice(sizeof(int)), ref curr, end))
+                                ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+                            noOfFoundChannels++;
+                        }
+                    }
+                }
+
+                if (noOfFoundChannels == 0)
+                {
+                    curr = ptr;
+                    while (!RespWriteUtils.WriteEmptyArray(ref curr, end))
+                        ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+                    return;
+                }
+
+                var newTotalArrayHeaderLen = 0;
+                var _ptr = ptr;
+                RespWriteUtils.WriteArrayLength(noOfFoundChannels, ref _ptr, end, out var _, out newTotalArrayHeaderLen); // ReallocateOutput is not needed here as there should be always be available space in the output buffer as we have already written the max array length
```

Comment:
_ = RespWriteUtils.WriteArrayLength to make it explicit that we are ignoring the return value

Reply:
Done